File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/MySideAmountFunction.java
package cz.fb.manaus.reactor.profit.progress.function;

import cz.fb.manaus.core.model.RunnerPrices;
import cz.fb.manaus.core.model.SettledBet;
import org.springframework.stereotype.Component;

@Component
public class MySideAmountFunction extends AbstractOfferedAmountFunction {

    @Override
    protected RunnerPrices getRunnerPrices(SettledBet bet) {
        var marketPrices = bet.getBetAction().getMarketPrices();
        var prices = marketPrices.getRunnerPrices(bet.getSelectionId());
        var side = bet.getPrice().getSide();
        return prices.getHomogeneous(side);
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/MySideAmountFunction.kt
package cz.fb.manaus.reactor.profit.progress.function

import cz.fb.manaus.core.model.RealizedBet
import cz.fb.manaus.core.model.RunnerPrices
import cz.fb.manaus.reactor.price.getRunnerPrices
import org.springframework.stereotype.Component

@Component
object MySideAmountFunction : AbstractOfferedAmountFunction() {

    override fun getRunnerPrices(bet: RealizedBet): RunnerPrices {
        val marketPrices = bet.betAction.runnerPrices
        val prices = getRunnerPrices(marketPrices, bet.settledBet.selectionId)
        val side = bet.settledBet.price.side
        return prices.by(side)
    }

}
